### PR TITLE
Fail file permissions FT-5149

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala [hunk 1]
```diff
@@ -133,6 +139,16 @@ class FileSystemFileCollector extends FileCollector[Try] {
       .map(Paths.get(_))
   }
 
+  private def checkPermissions(directory: File, files: Set[Path]): CheckedFiles = {
+    files.map(path => (path, directory / path.toString)).foldLeft(CheckedFiles(Set.empty[Path], Set.empty[Path])) {
+      case (checkedFiles, (path, file)) if file.isRegularFile && file.testPermission(PosixFilePermission.OTHERS_READ) =>
+        checkedFiles.copy(readableFiles = checkedFiles.readableFiles + path)
+      case (checkedFiles, (path, file)) =>
+        logger.error(s"Could not read file $file, make sure it is readable by everybody.")
```
Comment (xplosunn) [inline src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala]: This is not an error. It is expected to happen.

File: src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala [hunk 2]
```diff
@@ -9,14 +10,17 @@ import com.codacy.analysis.cli.clients.api.{FilePath, PathRegex, ProjectConfigur
...
 
 import scala.util.Try
 
-final case class FilesTarget(directory: File, files: Set[Path])
+final case class FilesTarget(directory: File, readableFiles: Set[Path], unreadableFiles: Set[Path])
 
 class FileSystemFileCollector extends FileCollector[Try] {
 
-  // TODO: Check if File will work or if we might need Path to support relative paths
+  private case class CheckedFiles(unreadableFiles: Set[Path], readableFiles: Set[Path])
```
Comment (xplosunn) [inline src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala]: Should be consistent with FilesTarget (readable first and unreadable after or vice versa)

File: src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala [hunk 3]
```diff
@@ -138,6 +145,13 @@ class AnalyseExecutor(toolInput: Option[String],
     }
   }
 
+  private def overrideFilePermissions(sourceDirectory: File) = {
+    Process(Seq("find", sourceDirectory.pathAsString, "-type", "d", "-exec", "chmod", "u+rwx", "{}", ";")).!
+    Process(Seq("find", sourceDirectory.pathAsString, "-type", "d", "-exec", "chmod", "ugo+rx", "{}", ";")).!
+    Process(Seq("find", sourceDirectory.pathAsString, "-type", "f", "-exec", "chmod", "u+rw", "{}", ";")).!
+    Process(Seq("find", sourceDirectory.pathAsString, "-type", "f", "-exec", "chmod", "ugo+r", "{}", ";")).!
+  }
```
Comment (pedrocodacy) [inline src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: Maybe move this to codacy-analysis-cli/src/main/scala/com/codacy/analysis/cli/utils/FileHelper.scala's FileHelper object

Comment (rtfpessoa) [reply src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: I do not want this to be used by anyone else neither in other places.

File: src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala (hunk 1 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala]: It is not expected to happen. The next phase is to fail the analysis due to this.
I was thinking a flag that can for can continuing after the one that forces permissions.

**Pull request merged**